openwisp_firmware_upgrader/tests/test_admin.py

```diff
@@ -319,6 +320,23 @@ def test_save_device_with_deleted_devicefirmware(self):
         )
         self.assertEqual(response.status_code, 200)
 
+    @capture_stderr()
+    def test_device_firmware_upgrade_without_device_connection(
```

Comment:
This test is ineffective. I added this test on the current master and it passes anyway.

Reply:
@nemesifier I copied this test in the master branch and it failed 

```
Found 1 test(s).
Creating test database for alias 'default'...
System check identified no issues (0 silenced).
F
Summary of slow tests (>0.3s)


Total slow tests detected: 0

======================================================================
FAIL: test_device_firmware_upgrade_without_device_connection (openwisp_firmware_upgrader.tests.test_admin.TestAdmin)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pandafy/openwisp/venv-firmware/lib/python3.8/site-packages/openwisp_utils/tests.py", line 99, in wrapped_function
    function(*args, **kwargs)
  File "/home/pandafy/openwisp/openwisp-firmware-upgrader/openwisp_firmware_upgrader/tests/test_admin.py", line 334, in test_device_firmware_upgrade_without_device_connection
    self.assertNotIn(
AssertionError: "'NoneType' object has no attribute 'update_strategy'" unexpectedly found in '\'NoneType\' object has no attribute \'update_strategy\'\nTraceback (most recent call last):\n  File "/home/pandafy/openwisp/openwisp-firmware-upgrader/openwisp_firmware_upgrader/utils.py", line 36, in get_upgrader_class_from_device_connection\n    upgrader_class = app_settings.UPGRADERS_MAP[device_conn.update_strategy]\nAttributeError: \'NoneType\' object has no attribute \'update_strategy\'\n\'NoneType\' object has no attribute \'update_strategy\'\nTraceback (most recent call last):\n  File "/home/pandafy/openwisp/openwisp-firmware-upgrader/openwisp_firmware_upgrader/utils.py", line 36, in get_upgrader_class_from_device_connection\n    upgrader_class = app_settings.UPGRADERS_MAP[device_conn.update_strategy]\nAttributeError: \'NoneType\' object has no attribute \'update_strategy\'\n'

----------------------------------------------------------------------
Ran 1 test in 3.125s

FAILED (failures=1)
Destroying test database for alias 'default'...
```